Sources/AppcuesKit/Appcues.swift

```diff
@@ -318,6 +312,25 @@ public class Appcues: NSObject {
         _ = container.resolve(UIKitScreenTracker.self)
     }
 
+    /// Enables automatic push notification management.
+    ///
+    /// This should be called in `UIApplicationDelegate.application(_:didFinishLaunchingWithOptions:)` to ensure no incoming notifications are missed.
+    ///
+    /// The following will automatically be handled:
+    /// 1. Calling `UIApplication.registerForRemoteNotifications()`
+    /// 2. Implementing `UIApplicationDelegate.application(_:didRegisterForRemoteNotificationsWithDeviceToken:)`
+    /// to call ``setPushToken(_:)``
+    /// 3. Ensuring `UNUserNotificationCenter.current().delegate` is set
+    /// 4. Implementing `UNUserNotificationCenterDelegate.userNotificationCenter(_:didReceive:withCompletionHandler:)`
+    /// to call ``didReceiveNotification(response:completionHandler:)``
+    /// 5. Implementing `UNUserNotificationCenterDelegate.userNotificationCenter(_:willPresent:withCompletionHandler:)`
+    /// to show notification while the app is in the foreground
+    @objc
+    public func enableAutomaticPushConfig() {
```

Comment:
Note that this is a function on `Appcues` instead of an `Appcues.Config` option because,

1. It’s easier to explain in the documentation that this should be called from `AppDelegate.didFinishLaunchingWithOptions` otherwise “assigning a delegate after the system calls these methods might cause you to miss incoming notifications.” A config option would mean being more prescriptive about how the `Appcues` instance gets inited.
2. It matches `Appcues.trackScreens()` above.